HW_2.py

```diff
@@ -0,0 +1,164 @@
+from pydantic import BaseModel
+from contextlib import contextmanager
+from abc import ABC, abstractmethod
+
```

Comment:
Due to PEP-8 rules there should be 2 line indent between classes and functions. Please user ruff, black or flake8 to check all PEP-8 rules and fix then